simplegrid/deep_cow.py

```diff
@@ -27,20 +27,36 @@ def __init__(self, x, y, energy, color=None):
     def to_internal_state(observation):
         """Convert state to an internal representation.
 
-        The input state is a NxN matrix with for each cell either a 1 for food,
-        0 for nothing or -x for another animal. The center cell is always "us"
+        The input state is a (2 x d + 1, 2 x d + 1) matrix with for each
+        cell either a 1 for food, 0 for nothing or -x for another animal.
+        The center cell is always "us". Only the largest diamond fitting
+        the matrix is actually visible.
         """
-
-        grass = np.copy(observation).flatten()
+        size = observation.shape[0]
+
+        view_distance = size // 2
+        if view_distance == 1:
+            diamond = observation.flatten()
+            diamond = [diamond[3], diamond[7], diamond[5], diamond[1]]
+        else:
+            size = observation.shape[0]
+            diamond = []
+            for x in range(size):
+                for y in range(size):
+                    if 0 < abs(x - size // 2) + abs(y - size // 2) <= view_distance:
+                        diamond.append(observation[x][y])
+        diamond = np.asarray(diamond)
```

Comment:
Why doesn't above code word for view_distance==1?

Reply:
it does work, but it would change the current behavior, which would break any previously saved models. Als the mapping of direction and cell breaks. We might not care, but for now I thought I keep it around.